agent/consul/catalog_endpoint.go

```diff
@@ -28,7 +28,7 @@ func (c *Catalog) Register(args *structs.RegisterRequest, reply *struct{}) error
 	defer metrics.MeasureSince([]string{"catalog", "register"}, time.Now())
 
 	// Verify the args.
-	if args.Node == "" || args.Address == "" {
+	if args.Node == "" || (args.Address == "" && !args.SkipNodeUpdate) {
```

Comment:
This is worth a unit test as well, so it doesn't break later (the impl. of the skip is pretty subtle).